### PR TITLE
Feature/151168140/game group chat

### PR DESCRIPTION
**Pull request by nebanat**
#### What does this PR do?
Users in a game are able to chat with each other 
#### Description of Task to be completed?
- When users are in a game, a chat messaging box appears on the left bottom corner of the window 
- Users can send messages between each other
- Only users on the game instance can see messages
#### How should this be manually tested?
Start a game then invite other users. Send messages to any users in the game
#### Any background context you want to provide?
Used socket.io for real-time chat message exchange between game users. Persisted chat threads in firebase (A NoSQL as a service) because it provides three-way binding with angular so that messages are saved real time.
#### What are the relevant pivotal tracker stories?
- Story Id: 151168140
- Story type: Feature
- Story: Users should be able to chat with the group while they are in a game 
#### Screenshots (if appropriate)
screen shot 2017-10-07 at 8 34 26 pm
screen shot 2017-10-07 at 8 34 18 pm
screen shot 2017-10-07 at 8 34 11 pm
screen shot 2017-10-07 at 8 33 42 pm

#### Questions:

### CODE REVIEW

File: config/socket/socket.js [hunk 1]
```diff
@@ -1,26 +1,53 @@
...
+/* eslint-disable */
+const Game = require('./game');
+const Player = require('./player');
+require('console-stamp')(console, 'm/dd HH:MM:ss');
+const mongoose = require('mongoose');
 
-var avatars = require(__dirname + '/../../app/controllers/avatars.js').all();
+const User = mongoose.model('User');
+const firebase = require('firebase');
+const fbref = require('../firebase/firebase.js');
```
Comment (ibonly) [inline config/socket/socket.js]: Change this variable name

File: config/socket/socket.js [hunk 2]
```diff
@@ -1,26 +1,53 @@
...
+// var admin = require('firebase-admin');
```
Comment (ibonly) [inline config/socket/socket.js]: Remove unused code

File: public/js/services/game.js [hunk 3]
```diff
@@ -223,11 +230,12 @@ angular.module('mean.system')
   };
 
   game.startGame = function() {
-
+    //console.log(game);
```
Comment (segunolalive) [inline public/js/services/game.js]: remove dead code

File: public/js/services/game.js [hunk 4]
```diff
@@ -242,6 +250,13 @@ angular.module('mean.system')
     socket.emit('pickWinning',{card: card.id});
   };
 
+  // game.messages = function(){
```
Comment (segunolalive) [inline public/js/services/game.js]: remove dead code

**Review by segunolalive — CHANGES_REQUESTED** [review]

File: public/js/services/game.js (hunk 3 above)
Comment (gbengaoyetade) [inline public/js/services/game.js]: remove redundant code

File: public/js/controllers/game.js [hunk 5]
```diff
@@ -1,10 +1,14 @@
 /*eslint-disable */
```
Comment (gbengaoyetade) [inline public/js/controllers/game.js]: Consider removing this

**Review by gbengaoyetade — COMMENTED** [review]
Remove redundant codes

File: public/js/controllers/game.js (hunk 5 above)
Comment (nebanat) [reply public/js/controllers/game.js]: Alright will do

**Review by ibonly — APPROVED** [review]

**Pull request merged**